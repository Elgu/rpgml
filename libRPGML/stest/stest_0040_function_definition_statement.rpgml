
Function f1()
{
  return 23;
}
print( f1() );
print( "\n" );

Function f2( d )
{
  return d;
}
print( f2( 0.75 ) );
print( "\n" );

Function f3( a, b, c="foo", d="bar" )
{
  print( a );
  print( " " );
  print( b );
  print( " " );
  print( c );
  print( " " );
  print( d );
  print( "\n" );
  return "42\n";
}
print( f3( 1, 2 ) );
print( f3( 1, 2, 3 ) );
print( f3( 1, 2, 3, 4 ) );
print( f3( c=1, a=2, d=3, b=4 ) );

Function F4( text="" )
{
  string text = text;
  Method getText() { return text; }
  Method setText( text ) { this.text = text; return this; }
  return this;
}
print( F4( "f4\n" ).getText() );
print( F4().setText( "bar\n" ).getText() );


