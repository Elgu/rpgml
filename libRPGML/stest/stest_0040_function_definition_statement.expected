23
0.75
1 2 foo bar
42
1 2 3 bar
42
1 2 3 4
42
2 4 1 3
42
f4
bar
